pkg/crc/cache/cache.go

```diff
@@ -1,39 +1,41 @@
 package cache
 
 import (
-	"io/ioutil"
-	"os"
-	"path/filepath"
-
+	"fmt"
 	"github.com/code-ready/crc/pkg/crc/constants"
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/download"
 	"github.com/code-ready/crc/pkg/embed"
 	"github.com/code-ready/crc/pkg/extract"
 	crcos "github.com/code-ready/crc/pkg/os"
 	"github.com/pkg/errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 )
 
 type Cache struct {
 	binaryName string
 	archiveURL string
 	destDir    string
+	version    string
+	getVerison func() (string, error)
 }
 
-func New(binarName string, archiveURL string, destDir string) *Cache {
-	return &Cache{binaryName: binarName, archiveURL: archiveURL, destDir: destDir}
+func New(binarName string, archiveURL string, destDir string, version string, getVerison func() (string, error)) *Cache {
+	return &Cache{binaryName: binarName, archiveURL: archiveURL, destDir: destDir, version: version, getVerison: getVerison}
 }
 
-func NewOcCache(destDir string) *Cache {
-	return New(constants.OcBinaryName, constants.GetOcUrl(), destDir)
+func NewOcCache(destDir string, version string, getVersion func() (string, error)) *Cache {
```

Comment:
@cfergeau version is something get from the constant but getVersion is not something constant and need to be passed. Even for version assume in case of libvirt driver to get that constant we need to import the libvrit driver constant to cache which doesn't look good. so I will keep it as it is for now.

Reply:
>  we need to import the libvrit driver constant to cache which doesn't look good.
 
The miscellaneous specialized OcCache, PodmanCache, ... already don't look that great in the generic cache package, so I would not mind getting the constant from where we need it. These specialized caches could be split off in their own file to 'hide' these dependencies from the main cache file :)

Reply:
And I don't think it's that great to export a method with arguments which are always going to be the same...
```
pkg/crc/preflight/preflight_checks_common.go:   oc := cache.NewOcCache(version.GetOcVersion(), getCurrentOcversion)
pkg/crc/preflight/preflight_checks_common.go:   oc := cache.NewOcCache(version.GetOcVersion(), getCurrentOcversion)
```
This just makes the method more complicated to use, the occache package is harder to understand as you need to look in multiple places for a clear image, ...

Reply:
I don't think we ever reached a consensus on this one...